### PR TITLE
Correction in documentation and Resolves #16

### LINKED ISSUE #16: Encode data as part of parse() method
Encode data as part of `parse()` method so it does not have to be called explicitly. Call `encode()` method inside `parse()` to accomplish this.

### PR DESCRIPTION
**Pull request by gsantoscomp**

### CODE REVIEW

File: src/CSV_Parser.php [hunk 1]
```diff
@@ -359,7 +361,7 @@ public function parse($headers = true) {
 	 * @throws Jabran\Exception\InvalidEncodingException
 	 * @return self
 	 */
-	public function encode() {
+	private function encode() {
```
Comment (jabranr) [inline src/CSV_Parser.php:364]: What is the reason for this change?

**Review by jabranr — COMMENTED** [review]
Thank you. I have one question for you though that I have left at the concerning line of code.

File: src/CSV_Parser.php (hunk 1 above)
Comment (gsantoscomp) [reply src/CSV_Parser.php:364]: Reading your issue, I had understood that the encode method was not to be accessed otherwise than by the parse method. I'm sorry if I misunderstood.

Comment (jabranr) [reply src/CSV_Parser.php:364]: Originally I kept it public so to be covered directly via unit tests and can be used for any reason if needed by a consumer. But I see that it can be a privately scoped and then tested via `parse()` method. Would you like to make all those changes or create a new issue of this and revert this scope change?

Comment (jabranr) [reply src/CSV_Parser.php:364]: Are we going ahead anymore on this?

**Comment by jabranr** [comment]
Closing this as there is no progress on it.
